src/fvm_multicell.hpp

```diff
@@ -628,9 +638,17 @@ void fvm_multicell<T, I>::advance(T dt) {
         auto ie = stim.second.amplitude(t_); // [nA]
         auto loc = stim.first;
 
+        // TODO KERNEL
+        // is a kernel actually needed?
+        // for now I only make the update if the injected current in nonzero to
+        // avoid a redundant host->device copy on the gpu
+        //
         // note: current_ in [mA/cm^2], ie in [nA], cv_areas_ in [µm^2].
         // unit scale factor: [nA/µm^2]/[mA/cm^2] = 100
-        current_[loc] -= 100*ie/cv_areas_[loc];
+        //current_[loc] -= 100*ie/cv_areas_[loc];
+        if (ie!=0.) {
+            current_[loc] = current_[loc] - 100*ie/cv_areas_[loc];
```

Comment:
Yes: the types returned by `operator[]` for `DeviceVector` are proxies, which are converted to `value_type` on asignment. Operators like `-=` have not been defined for these proxies.

This isn't a problem any more, because the gpu and cpu code for this loop has now been refactored into target specific loops/kernels.